File: Leetcode/sbgo/Leetcode419-甲板上的战舰.go
package sbgo

func countBattleships(board [][]byte) int {
	ans, n, m := 0, len(board), len(board[0])
	for i := 0; i < n; i++ {
		for j := 0; j < m; j++ {
			if i > 0 && board[i-1][j] == 'X' {
				continue
			}
			if j > 0 && board[i][j-1] == 'X' {
				continue
			}
			if board[i][j] == 'X' {
				ans++
			}
		}
	}
	return ans
}

File: Leetcode/Rust/Leetcode419-甲板上的战舰.rs
impl Solution {
    pub fn count_battleships(board: Vec<Vec<char>>) -> i32 {
        let (mut ans,n,m) = (0,board.len(),board[0].len());
        for i in 0..n {
            for j in 0..m {
                if i > 0 && board[i - 1][j] == 'X' {
                    continue;
                }
                if j > 0 && board[i][j - 1] == 'X' {
                    continue;
                }
                if board[i][j] == 'X' {
                    ans += 1;
                }
            }
        }
        ans
    }
}